lib/dao.js

```diff
@@ -1413,11 +1413,17 @@ DataAccessObject.removeById = DataAccessObject.destroyById = DataAccessObject.de
 
   var Model = this;
 
-  this.remove(byIdQuery(this, id).where, options, function(err) {
+  this.remove(byIdQuery(this, id).where, options, function(err, data) {
+    var deleted = data && data.count > 0;
     if ('function' === typeof cb) {
-      cb(err);
+      if (options.reportFailureAsError && !err && !deleted) {
+        err = new Error('No instance with id ' + id + ' found for ' + Model.modelName);
+        cb(err, false);
+      } else {
+        cb(err, deleted);
+      }
     }
-    if(!err) Model.emit('deleted', id);
+    if(!err && deleted) Model.emit('deleted', id);
```

Comment:
This is IMO a breaking change.

I am proposing to use the same flag that is controlling whether `count==0` triggers an error to control this behaviour. In fact, when that flag is set and no records were deleted, then this branch is never executed and thus this line does not need any change.